gh-weblog/components/Editor.jsx

```diff
@@ -12,25 +13,30 @@ export default createClass({
   render() {
     return (
       <textarea
-        ref="textarea"
+        ref={this.refs.textarea}
         className="editor"
         hidden={this.props.hidden}
         value={this.state.text}
-        onChange={this.record}
+        onInput={this.record}
       />
     );
   },
 
   setText(text) {
-    const textarea = this.refs.textarea.getDOMNode();
+    const textarea = this.refs.textarea.current;
     this.setState({ text }, () => textarea.focus());
   },
 
   record(evt) {
     const postData = evt.target.value;
     this.setState({ text: postData });
-    const [title] = postData.match(/^#\s+[^\n]+\n+/);
-    postData.replace(title, ``);
+    const titleMatch = postData.match(/^#\s+([^\n]+)\n+/);
+    let title = "";
+    if (titleMatch) {
+      const titleLine = titleMatch[0];
+      title = titleMatch[1];
+      postData.replace(titleLine, ``);
+    }
```

Comment:
This is doing way more work than necessary given that my posts always come with a title (because the initial new post content already includes one)

Reply:
Is there a way to set the initial content? For me, it was blank and I got an error in the console because `postData.match(...)` is null on a non-titled post.

Reply:
It was blank.... when? If you're authed you get the "new post" button, and clicking that calls [createEntry](https://github.com/Pomax/pomax.github.io/blob/main/gh-weblog/components/WebLog.jsx#L291-L304), which prepopulates the post.